spec/models/course/level_spec.rb

```diff
@@ -2,11 +2,55 @@
 
 RSpec.describe Course::Level, type: :model do
   it { is_expected.to belong_to(:course).inverse_of(:levels) }
-  it { is_expected.to validate_numericality_of(:experience_points_threshold).is_greater_than(0) }
+  it 'ensures that experience points threshold is greater or equal to 0' do
+    expect(subject).
+      to validate_numericality_of(:experience_points_threshold).
+      is_greater_than_or_equal_to(0)
+  end
+
+  let!(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    let!(:course) { build(:course) }
+
+    context 'when new level have the same threshold as existing level' do
+      before { create(:course_level, course: course, experience_points_threshold: 100) }
+      subject { build(:course_level, course: course, experience_points_threshold: 100) }
+
+      it 'is invalid' do
+        expect(subject.valid?).to be_falsey
+      end
+    end
+
+    context 'before level_number is set' do
+      it 'raises an IllegalStateError' do
+        expect { Course::Level.new.level_number }.to raise_error(IllegalStateError)
+      end
+    end
+
+    describe '.after_course_initialize' do
+      it 'builds only one default level' do
```

Comment:
I would split this to:
 - It builds a default level
 - context 'when the course is initialized again' it 'does not build another level'